lib/xboxrt/assert.h

```diff
@@ -0,0 +1,11 @@
+#ifndef __ASSERT_H
+#define __ASSERT_H
+
+#ifndef NDEBUG
+#include "debug.h"
+#define assert(x) if (!(x)) { debugPrint("\nAssertion failed! '%s' in %s:%d\n\n", #x, __FILE__, __LINE__); }
```

Comment:
nit: I don't like our `debugPrint` solution because we should not expect a framebuffer, serial connection, ... . I think we should force users to provide their own `debugPrint` handlers (and the default should be a no-operation function)

I'm fine with keeping this for now though, it's not your fault and an overall problem in nxdk.